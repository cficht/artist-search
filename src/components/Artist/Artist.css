.Artist {
  padding: 10px 10px 10px 10px;

  background-color: red;
  color: white;
  text-align: center;

  ul {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;

    margin-block-start: 0em;
    padding-inline-start: 0px;

    overflow: auto;
    list-style-type: none;
  }

  li {
    float: left;

    padding: 10px;
    margin: 10px;
    width: 17%;
    border-radius: 10px;

    background-color: darkblue;

    text-align: center;
  }

  img {
    height: 200px;
    width: 200px;
    margin: 10px;
  }

  a {
    color: white;
    font-size: 16px;
  }
}