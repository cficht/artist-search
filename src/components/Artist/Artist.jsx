import React from 'react';
import PropTypes from 'prop-types';
import styles from './Artist.css';

const Artist = ({ artistName, releases }) => {
  const releaseNodes = releases.map(release => (
    <li key={release.id}> 
      <img src={release.art} />
      <h3>{release.title}</h3>
    </li>  
  ));

  return (
    <section className={styles.Artist}>
      <h2>{artistName}</h2>
      <ul>
        {releaseNodes}
      </ul>
    </section>
  );
};

Artist.propTypes = {
  artistName: PropTypes.string.isRequired,
  releases: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  })).isRequired
};

export default Artist;
