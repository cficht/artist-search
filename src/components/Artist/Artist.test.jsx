import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artist component', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist 
      releases={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
