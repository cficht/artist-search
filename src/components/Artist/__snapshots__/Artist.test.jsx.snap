// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders Artist 1`] = `
<section
  className="Artist"
>
  <h2>
    The Beatles
  </h2>
  <ul />
</section>
`;
